File: Module 1/More_Examples/3_Object_Oriented_Example/Python/qubit.py
class Qubit:
    def __init__(self, u, d):
        self.up = u
        self.down = d

    def X(self):
        temp = self.up
        self.up = self.down
        self.down = temp

    def CX(self, control):
        if control.down == 1:
            self.X()

    def print(self):
        print("Spin Up: ", self.up)
        print("Spin Down: ", self.down)


if __name__ == "__main__":
    q1 = Qubit(1, 0)
    q2 = Qubit(0, 1)

    q1.CX(q2)

    q1.print()
    q2.print()

File: Module 1/More_Examples/3_Object_Oriented_Example/Rust/qubit.rs
fn main() {
    let mut _q1:Qubit = Qubit::new(1,0);
    _q1.X();
    println!("  Spin Up: {}", _q1.up);
    println!("Spin Down: {}", _q1.down);
}

struct Qubit {
      up: i64,
    down: i64
}

impl Qubit {
    fn new(up: i64, down: i64) -> Qubit {
        Qubit {
              up: up,
            down: down
        }
    }

    fn X(&mut self){
        let temp: i64 = self.up;
        self.up = self.down;
        self.down = temp;
    }

}